src/Money.php

```diff
@@ -78,6 +78,24 @@ public function __construct($amount, Currency $currency)
         $this->currency = $currency;
     }
 
+    /**
+     * @param object|array $normalized
+     *
+     * @return Money
+     *
+     * @throws \InvalidArgumentException when the normalized version cannot be denormalized
+     */
+    public static function denormalize($normalized)
+    {
+        if (is_object($normalized) && isset($normalized->amount) && isset($normalized->currency)) {
+            return new Money($normalized->amount, new Currency($normalized->currency));
+        } elseif (is_array($normalized) && isset($normalized['amount']) && isset($normalized['currency'])) {
```

Comment:
If on a new line.

Reply:
I guess you could also just cast the array to stdClass using `(object)` or the other way around and get rid of half the lines.